packages/okta-react/src/Auth.js

```diff
@@ -71,13 +71,27 @@ export default class Auth {
   }
 
   async getIdToken() {
-    const idToken = await this._oktaAuth.tokenManager.get('idToken');
-    return idToken ? idToken.idToken : undefined;
+    try {
+      const idToken = await this._oktaAuth.tokenManager.get('idToken');
+      return idToken.idToken;
+    } catch (err) {
+      // The user no longer has an existing SSO session in the browser.
+      // (OIDC error `login_required`)
+      // Ask the user to authenticate again.
+      return undefined;
```

Comment:
Just wondering, is this the only error that will be returned if promise is rejected?

The implementation pretty much says if a promise is rejected, regardless of what it is, ask the user to authenticate again (by returning `undefined).

Reply:
Just wondering, is this the only error that will be returned if promise is rejected?

The implementation pretty much says if a promise is rejected, regardless of what it is, ask the user to authenticate again (by returning `undefined).

Wouldn't it be better to *catch* errors that you know and then reject the rest that you don't know how to *catch* yet?

```js
try { }
catch(error) {
  if(error.type === 'login_required') {
    return handleLoginRequiredError(error); // returns `undefined`
  }

  return Promise.reject(error);
}
```

Or you could use an `errorHandlerMap` if there's more than one error that you know.

Reply:
Hi @rodoabad , good question.  You're right that the current approach is opinionated.  If this ends up being too limiting, please let us know and we'll make the error more workable.

Reply:
@robertjd @rodoabad @jmelberg-okta @manueltanzi-okta 

How do I log the user out after a session has expired? 

In my Okta org I set tokens to expire after 5 minutes, and set sessions to expire after 6 minutes. When I am testing, I attempt to access a `<SecureRoute>` after the session has expired. The secure route is loaded (even though the session has expired), and inside the component `auth.isAuthenticated()` is called, which internally catches the `login_required` error and returns `undefined`. At this point, if I call `auth.logout()`, I get an error:
```
Unhandled Rejection (AuthApiError): Not found: Resource not found: me (Session)
```

So, should I just redirect to my `/login` path even though the auth module does not consider me to be logged out? What's the way I should be approaching this? I just want my application state to reflect the fact that my Okta session has ended.